api4/team.go

```diff
@@ -235,6 +236,60 @@ func getTeamMembersByIds(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(model.TeamMembersToJson(members)))
 }
 
+func addTeamMember(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+
+	var tm *model.TeamMember
+	var err *model.AppError
+	var team *model.Team
+	if team, err = app.GetTeam(c.Params.TeamId); err != nil {
+		c.Err = err
+		return
+	}
```

Comment:
The request body should be `teamMember.ToJson()` which you can create with `teamMember = &model.TeamMember{TeamId: teamId, UserId: userId}`.

Also we should still check `member.TeamId == c.Params.TeamId` since we could get requests that don't come from the driver and they might mismatch the team id in the url and the one in the team member